moebot_bot/bot/commands/timer.go

```diff
@@ -0,0 +1,71 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/camd67/moebot/moebot_bot/bot/permissions"
+	"github.com/camd67/moebot/moebot_bot/util"
+	"github.com/camd67/moebot/moebot_bot/util/db"
+)
+
+type TimerCommand struct {
+	chTimers util.SyncChannelTimerMap
+	Checker  permissions.PermissionChecker
+}
+
+func NewTimerCommand() *TimerCommand {
+	tc := &TimerCommand{}
+	tc.chTimers = util.SyncChannelTimerMap{
+		RWMutex: sync.RWMutex{},
+		M:       make(map[string]time.Time),
+	}
+	return tc
+}
+
+func (tc *TimerCommand) Execute(pack *CommPackage) {
+	channelID := pack.message.ChannelID
+	if len(pack.params) > 0 && strings.EqualFold(pack.params[0], "start") {
+		if tc.Checker.HasPermission(pack.message.Author.ID, pack.member.Roles, pack.guild, db.PermMod) {
+			tc.chTimers.Lock()
+			tc.chTimers.M[channelID] = time.Now()
+			tc.chTimers.Unlock()
+			pack.session.ChannelMessageSend(pack.message.ChannelID, "Timer started! OwO")
```

Comment:
Ah  I see what you mean by emoticons now...

Unfortunately I'm going to request these removed. I'd prefer to keep moebot usable by any server, even those who aren't particularly into anime. While a lot of features are tailored to anime communities, a timer command certainly isn't.

All that being said, I just thought of the idea of having different "modes" you can switch on within moebot that use different strings for this kind of stuff... I'll add that as a separate feature